src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -83,5 +86,28 @@ protected override CompletionItemRules GetCompletionItemRules()
         {
             return ItemRules.Instance;
         }
+
+        protected override async Task<IEnumerable<ISymbol>> GetPreselectedSymbolsWorker(AbstractSyntaxContext context, int position, OptionSet options, CancellationToken cancellationToken)
+        {
+            var recommender = context.GetLanguageService<IRecommendationService>();
+            var typeInferrer = context.GetLanguageService<ITypeInferenceService>();
+
+            var inferredTypes = typeInferrer.InferTypes(context.SemanticModel, position, cancellationToken)
+                ?.Where(t => t.SpecialType != SpecialType.System_Void)
+                .ToSet();
+            if (inferredTypes == null || !inferredTypes.Any())
+            {
+                return SpecializedCollections.EmptyEnumerable<ISymbol>();
+            }
+
+            var symbols = await recommender.GetRecommendedSymbolsAtPositionAsync(context.Workspace, context.SemanticModel, position, options, cancellationToken).ConfigureAwait(false);
+            return symbols.Where(s => inferredTypes.Contains(s.GetSymbolType()) && !IsInstrinsic(s));
+        }
+
+        private bool IsInstrinsic(ISymbol s)
+        {
+            var ts = s as ITypeSymbol;
+            return ts != null && ts.SpecialType != SpecialType.None;
```

Comment:
The goal was to not preselect the type "Integer" but instead the keyword "int". SpecialType is clearly not appropriate. Unfortunately, the compiler's helper for this is internal so I'll have to make a copy in the IDE :(